src/Features/CSharp/Portable/CodeMapper/CSharpMapCodeService.Scope.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeMapping;
+
+internal sealed partial class CSharpMapCodeService
+{
+    /// <summary>
+    /// The level of the scope where this node is located.
+    /// NOTE: The scope does not represent where this node will be inserted.
+    /// The scope represents the hierarchy this node has.
+    /// For example, a Class Scope node means that it needs to be placed next to
+    /// other classes or interfaces, and not inside them.
+    /// NOTE: The order in which these scopes are setup on this enum matter.
+    /// They should go from lower to higher in terms of what goes inside what.
+    /// For example, Class is the highest scope here, because the class will usually contain methods, and methods cannot contain classes.
+    /// Same with statements.
+    /// </summary>
+    internal enum Scope
+    {
+        /// <summary>
+        /// Unknown scope.
+        /// </summary>
+        None,
```

Comment:
when does this happen?

Reply:
Whenever it is not set.
It shouldn't happen, but it's a practice I've done in the past to trump the default enum to be able to detect errors.